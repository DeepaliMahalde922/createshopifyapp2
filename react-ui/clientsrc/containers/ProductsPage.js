// @flow
import _ from 'lodash';
import React, { Component } from 'react';
import type { Element } from 'react';
import type { RouterHistory } from 'react-router-redux';
import { connect } from 'react-redux';

import { Page, Card, ResourceList, Thumbnail, Select, Layout, FormLayout, TextField, ChoiceList, Heading, Button, DisplayText } from '@shopify/polaris';
import { ResourcePicker } from '@shopify/polaris/embedded';

import { addProduct, fetchProducts } from '../redux/products';
import type { AddProductAction, Product, State, ThunkAction } from '../types';

type ProductResource = {
  media?: Element<*>,
  attributeOne: string,
  attributeTwo: string,
  attributeThree: string
};

type OwnProps = {
  history: RouterHistory
};

type StateProps = {
  products: ProductResource[]
};

type DispatchProps = {
  addProduct: (product: Product) => AddProductAction,
  fetchProducts: () => ThunkAction
};

type Props = StateProps & DispatchProps & OwnProps;

type OwnState = {
  resourcePickerOpen: boolean
};

type Resources = {
  products: Product[]
};

export class ProductsPageComponent extends Component<Props, OwnState> {
  state = {
    resourcePickerOpen: false
  };

  componentDidMount() {
    const { fetchProducts } = this.props;

    fetchProducts();
  }

  handleGoToProducts = () => {
    const { history } = this.props;

    history.push('/Review');
  };

  handleGoToOrderlist = () => {
    const { history } = this.props;

    history.push('/Orderlist');
  };

  handleGoToOrders = () => {
    const { history } = this.props;

    history.push('/Orders');
  };

  handleResourceSelected = (resources: Resources) => {
    const { addProduct } = this.props;
    const { products } = resources;

    addProduct(products[0]);

    this.setState({ resourcePickerOpen: false });
  };

  render() {
    const { products = [] } = this.props;
    const { resourcePickerOpen } = this.state;

    return (
      <Page
        title="Products"
        primaryAction={{
          content: 'Add product 122',
          onAction: () => this.setState({ resourcePickerOpen: true })
        }}
        secondaryActions={[
          {
            content: 'About',
            onAction: this.handleGoToProducts
          },
          {
            content: 'Orders',
            onAction: this.handleGoToOrders
          },
          {
            content: 'Orders List',
            onAction: this.handleGoToOrderlist
          }
        ]}
        
      >
<DisplayText size="extraLarge">ContentMark</DisplayText>        <ResourcePicker
          products
          open={resourcePickerOpen}
          onSelection={this.handleResourceSelected}
          onCancel={() => this.setState({ resourcePickerOpen: false })}
          allowMultiple={true}
        />
        <Layout sectioned>
        <Layout.Section>
        <Card>
          <ResourceList
            items={products}
            renderItem={(item: ProductResource, index: number) =>
              <ResourceList.Item key={index} {...item} />}
          />
        </Card>
        </Layout.Section>
        <Layout.Section>
        <Card sectioned
        title="Type of Content"
        >
            <ChoiceList
            
            choices={[
              {
                label: 'How To Article',
                value: 'howto',
                
              }
              ,
              {
                label: 'Best Of Article',
                value: 'experience',
              },
              {
                label: 'Required',
                value: 'required',
              },
            ]}
            selected={['required']}
          />
          
          </Card>
        </Layout.Section>
        <Layout.Section secondary>
          <Card
            title="Please describe what kind of article you would like written"
            sectioned
          >
            <TextField
  
  multiline
/>
          </Card>
        </Layout.Section>
        <Layout.Section secondary>
        <DisplayText size="small">The cost of the article is $50. You will have a chance to review and approve the article before submitting payment.</DisplayText>
</Layout.Section>
        <Layout.Section secondary>
        <Button
  size="large"
  primary
  submit
>
  Request Article
</Button>
</Layout.Section>
      </Layout>
      </Page>
    );
  }
}

const getProductResources = (products: ?(Product[])) =>
  _.map(products, (product: Product): ProductResource => {
    const { image = {}, product_type, title, vendor } = product;

    return {
      media: image && <Thumbnail source={image.src} alt={title} />,
      attributeOne: title,
      attributeTwo: product_type,
      attributeThree: vendor
    };
  });

const mapStateToProps = (state: State): StateProps => {
  const { products } = state;

  return {
    products: getProductResources(products)
  };
};

const dispatchProps: DispatchProps = { addProduct, fetchProducts };

export default connect(mapStateToProps, dispatchProps)(ProductsPageComponent);