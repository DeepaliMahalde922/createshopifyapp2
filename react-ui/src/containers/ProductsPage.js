// @flow
import _ from 'lodash';
import React, { Component } from 'react';
import type { Element } from 'react';
import type { RouterHistory } from 'react-router-redux';
import { connect } from 'react-redux';

import { Page, Card, ResourceList, Thumbnail } from '@shopify/polaris';
import { ResourcePicker } from '@shopify/polaris/embedded';

import { addProduct, fetchProducts } from '../redux/products';
import type { AddProductAction, Product, State, ThunkAction } from '../types';

type ProductResource = {
  media?: Element<*>,
  attributeOne: string,
  attributeTwo: string,
  attributeThree: string
};

type OwnProps = {
  history: RouterHistory
};

type StateProps = {
  products: ProductResource[]
};

type DispatchProps = {
  addProduct: (product: Product) => AddProductAction,
  fetchProducts: () => ThunkAction
};

type Props = StateProps & DispatchProps & OwnProps;

type OwnState = {
  resourcePickerOpen: boolean
};

type Resources = {
  products: Product[]
};

export class ProductsPageComponent extends Component<Props, OwnState> {
  state = {
    resourcePickerOpen: false
  };

  componentDidMount() {
    const { fetchProducts } = this.props;

    fetchProducts();
  }

  handleGoToProducts = () => {
    const { history } = this.props;

    history.push('/about');
  };

  handleResourceSelected = (resources: Resources) => {
    const { addProduct } = this.props;
    const { products } = resources;

    addProduct(products[0]);

    this.setState({ resourcePickerOpen: false });
  };

  render() {
    const { products = [] } = this.props;
    const { resourcePickerOpen } = this.state;

    return (
      <Page
        title="Products"
        primaryAction={{
          content: 'Add product',
          onAction: () => this.setState({ resourcePickerOpen: true })
        }}
        secondaryActions={[
          {
            content: 'Go to About',
            onAction: this.handleGoToProducts
          }
        ]}
      >
        <Card>
          <ResourceList
            items={products}
            renderItem={(item: ProductResource, index: number) =>
              <ResourceList.Item key={index} {...item} />}
          />
        </Card>
        <ResourcePicker
          products
          open={resourcePickerOpen}
          onSelection={this.handleResourceSelected}
          onCancel={() => this.setState({ resourcePickerOpen: false })}
        />
      </Page>
    );
  }
}

const getProductResources = (products: ?(Product[])) =>
  _.map(products, (product: Product): ProductResource => {
    const { image = {}, product_type, title, vendor } = product;

    return {
      media: image && <Thumbnail source={image.src} alt={title} />,
      attributeOne: title,
      attributeTwo: product_type,
      attributeThree: vendor
    };
  });

const mapStateToProps = (state: State): StateProps => {
  const { products } = state;

  return {
    products: getProductResources(products)
  };
};

const dispatchProps: DispatchProps = { addProduct, fetchProducts };

export default connect(mapStateToProps, dispatchProps)(ProductsPageComponent);
