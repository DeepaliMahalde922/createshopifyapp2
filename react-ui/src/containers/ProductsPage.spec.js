import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { ProductsPageComponent as ProductsPage } from './ProductsPage';

const fetchProducts = jest.fn(() => true);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ProductsPage fetchProducts={fetchProducts} />, div);
});

it('Matches snapshot', () => {
  const tree = renderer
    .create(<ProductsPage fetchProducts={fetchProducts} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
