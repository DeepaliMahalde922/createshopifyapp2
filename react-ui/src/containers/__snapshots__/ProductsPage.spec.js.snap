// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="Polaris-Page"
>
  <div
    className="Polaris-Page__Header Polaris-Page__Header--hasSecondaryActions"
  >
    <div
      className="Polaris-Page__MainContent"
    >
      <div
        className="Polaris-Page__TitleAndActions"
      >
        <div
          className="Polaris-Page__Title"
        >
          <h1
            className="Polaris-DisplayText Polaris-DisplayText--sizeLarge"
          >
            Products
          </h1>
        </div>
        <div
          className="Polaris-Page__Actions"
        >
          <div
            className="Polaris-Page__PrimaryAction"
          >
            <button
              aria-busy={undefined}
              aria-label={undefined}
              className="Polaris-Button Polaris-Button--primary"
              disabled={undefined}
              onBlur={undefined}
              onClick={[Function]}
              onFocus={undefined}
              onMouseUp={[Function]}
              role={undefined}
              type="button"
            >
              <span
                className="Polaris-Button__Content"
              >
                <span>
                  Add product
                </span>
              </span>
            </button>
          </div>
          <div
            className="Polaris-Page__SecondaryActions"
          >
            <div
              className="Polaris-Page__IndividualActions"
            >
              <button
                aria-label={undefined}
                className="Polaris-Page__Action"
                onClick={[Function]}
                onMouseUp={[Function]}
                type="button"
              >
                Go to About
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="Polaris-Page__PrimaryAction"
      >
        <button
          aria-busy={undefined}
          aria-label={undefined}
          className="Polaris-Button Polaris-Button--primary"
          disabled={undefined}
          onBlur={undefined}
          onClick={[Function]}
          onFocus={undefined}
          onMouseUp={[Function]}
          role={undefined}
          type="button"
        >
          <span
            className="Polaris-Button__Content"
          >
            <span>
              Add product
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="Polaris-Page__Content"
  >
    <div
      className="Polaris-Card"
    >
      <ul
        className="Polaris-ResourceList"
      />
    </div>
  </div>
</div>
`;
